membership-attribute-service/app/components/TouchpointComponents.scala

```diff
@@ -124,28 +117,21 @@ class TouchpointComponents(
   }
 
   lazy val catalogRestClient = rest.SimpleClient[Future](backendConfig.zuoraRest, RequestRunners.configurableFutureRunner(60.seconds))
-  lazy val catalogService = catalogServiceOverride.getOrElse(
-    new CatalogService(
-      productIds,
-      FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix),
-      Await.result(_: Future[Catalog], 60.seconds),
-      stage.value,
-    ),
+  lazy val futureCatalog: Future[CatalogMap] = catalogServiceOverride.getOrElse(
+    CatalogService
+      .read(FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix))
+      .recover { case error =>
+        logger.errorNoPrefix(scrub"Failed to load the product catalog from Zuora due to: $error")
+        throw error
+      },
   )
 
-  private lazy val futureCatalog: Future[CatalogMap] = catalogService.catalog
-    .map(_.fold[CatalogMap](error => { logger.errorNoPrefix(scrub"error: ${error.list.toList.mkString}"); Map() }, _.map))
-    .recover { case error =>
-      logger.errorNoPrefix(scrub"Failed to load the product catalog from Zuora due to: $error")
-      throw error
-    }
```

Comment:
this code was a bit convoluted so now it's just a small snippet that gives us a `Future[CatalogMap]` without as much fanfare

Reply:
Is it loading the catalog from S3? The name makes it sound like it's going to Zuora

Reply:
yes it does look like it's getting it from zuora (with a 60s timeout) 
I did wonder if the S3 catalog was originally for the benefit of lambdas, but I didn't look into it at this stage.  There's code there to read from S3 but it's not used, but I've kept it just in case.